cpp/src/io/utilities/config_utils.cpp

```diff
@@ -244,16 +244,20 @@ CUDF_EXPORT std::mutex& host_mr_mutex()
 }
 
 // Must be called with the host_mr_mutex mutex held
-CUDF_EXPORT rmm::host_async_resource_ref& make_host_mr(std::optional<host_mr_options> const& opts)
+CUDF_EXPORT rmm::host_async_resource_ref& make_host_mr(std::optional<host_mr_options> const& opts,
+                                                       bool* did_configure = nullptr)
```

Comment:
optional: it's preferable to return all results instead of having an out parameter. This is an implementation detail here so it's not a blocking concern IMO.

Reply:
I made this work by returning `std::pair<rmm::host_async_resource_ref*, bool>`. I'll push that shortly.

Reply:
Please see: https://github.com/rapidsai/cudf/pull/15815/commits/afd4cbe6f815faee36fcb977e0b86b866a257351